numpy/core/tests/test_scalarbuffer.py

```diff
@@ -0,0 +1,107 @@
+"""
+Test scalar buffer interface adheres to PEP 3118
+"""
+import sys
+import numpy as np
+from numpy.testing import run_module_suite, assert_, dec
+
+# types
+scalars = [
+    np.bool_, np.bool8, np.byte, np.short, np.intc, np.int_,
+    np.longlong, np.intp, np.int8, np.int16, np.int32, np.int64,
+    np.ubyte, np.ushort, np.uintc, np.uint, np.ulonglong,
+    np.uintp, np.uint8, np.uint16, np.uint32, np.uint64,
+    np.half, np.single, np.double, np.float_, np.longfloat,
+    np.float16, np.float32, np.float64, np.csingle, np.complex_,
+    np.clongfloat, np.complex64, np.complex128,
+]
+
+scalars_set_code = [
+    (np.bool_, '?'),
+    (np.byte, 'b'),
+    (np.short, 'h'),
+    (np.intc, 'i'),
+    (np.int_, 'l'),
+    (np.longlong, 'q'),
+    (np.ubyte, 'B'),
+    (np.ushort, 'H'),
+    (np.uintc, 'I'),
+    (np.uint, 'L'),
+    (np.ulonglong, 'Q'),
+    (np.half, 'e'),
+    (np.single, 'f'),
+    (np.double, 'd'),
+    (np.float_, 'd'),
+    (np.longfloat, 'g'),
+    (np.csingle, 'Zf'),
+    (np.complex_, 'Zd'),
+    (np.clongfloat, 'Zg'),
+
+    # TODO: 'p', 'l', or 'q'?
+    # (np.intp, 'p'),
+    # (np.uintp, 'P'),
+]
+
+scalars_set_size = [
+    (np.bool8, 1),
+    (np.int8, 1),
+    (np.int16, 2),
+    (np.int32, 4),
+    (np.int64, 8),
+    (np.uint8, 1),
+    (np.uint16, 2),
+    (np.uint32, 4),
+    (np.uint64, 8),
+    (np.float16, 2),
+    (np.float32, 4),
+    (np.float64, 8),
+    (np.complex64, 8),
+    (np.complex128, 16),
+]
+
+# platform dependant dtypes
+if hasattr(np, 'float96'):
+    scalars.append(np.float96)
+    scalars_set_size.append((np.float96, 12))
+if hasattr(np, 'float128'):
+    scalars.append(np.float128)
+    scalars_set_size.append((np.float128, 16))
+if hasattr(np, 'complex192'):
+    scalars.append(np.complex192)
+    scalars_set_size.append((np.complex192, 24))
+if hasattr(np, 'complex256'):
+    scalars.append(np.complex256)
+    scalars_set_size.append((np.complex256, 32))
+
+
+class TestScalarPEP3118(object):
+    @dec.skipif(sys.version_info.major < 3, "scalars do not implement buffer interface in Python 2")
+    def test_scalar_match_array(self):
+        for scalar in scalars:
+            x = scalar()
+            a = np.array([], dtype=scalar)
+            assert_(x.data.format == a.data.format)
+
+    @dec.skipif(sys.version_info.major < 3, "scalars do not implement buffer interface in Python 2")
```

Comment:
Not sure this is true - can you use `memoryview(x)` to get hold of it?

Reply:
Oddly, no.
```
Python 2.7.14 (default, Sep 23 2017, 22:06:14) 
[GCC 7.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> x = np.int16(2)
>>> memoryview(x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot make memory view because object does not have the buffer interface
>>> x.data
<read-only buffer for 0x7ff23b84d150, size -1, offset 0 at 0x7ff23b778430>
>>> 
```

Reply:
Hmm, that's pretty odd, given it works fine on arrays